src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,152 @@
+     *
+     * @param taskIdx The index of the task to be marked as done.
+     */
+    public void markTaskAsDone(int taskIdx) {
+        Task task = this.tasks.get(taskIdx);
+        task.markAsDone();
+    }
+
+    /**
+     * Creates a ToDo task.
+     *
+     * @param task The task description of the ToDo.
+     * @return A ToDO object.
+     */
+    public ToDo createTodo(String task) {
+        assert (task != null);
+        return new ToDo(task, false);
+    }
+
+    /**
+     * Creates a deadline.
+     *
+     * @param task The task description of the deadline.
+     * @return A DeadLine object.
+     * @throws UnrecognizableCommandException If the command is invalid.
+     * @throws InvalidDateTimeException       If the date is invalid.
+     */
+    public DeadLine createDeadLine(String task) throws UnrecognizableCommandException, InvalidDateTimeException {
+        assert (task != null);
+        if (!task.contains(" /by ")) {
+            throw new UnrecognizableCommandException();
+        }
+        try {
+            String[] lst = task.split(" /by ");
+            String description = lst[0];
+            String ddl = lst[1];
+            LocalDate date = parser.stringToLocalDate(ddl);
+            return new DeadLine(description, date, false);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new UnrecognizableCommandException();
+        }
+
+    }
+
+    /**
+     * Creates a event
+     *
+     * @param task The task description of the event.
+     * @return An event object.
+     * @throws UnrecognizableCommandException If the command is invalid.
+     * @throws InvalidDateTimeException       If the date is invalid.
+     */
+    public Event createEvent(String task) throws UnrecognizableCommandException, InvalidDateTimeException {
+        assert (task != null);
+        if (!task.contains(" /at ")) {
+            throw new UnrecognizableCommandException();
+        }
+        try {
+            String[] lst = task.split(" /at ");
+            String description = lst[0];
+            String period = lst[1];
+            LocalDate date = parser.stringToLocalDate(period);
+            return new Event(description, date, false);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new UnrecognizableCommandException();
+        }
+    }
+
+    /**
+     * Finds a list of matching tasks given some keywords.
+     *
+     * @param keywords Keywords to be found among all tasks.
+     * @return A list of matched tasks.
+     */
+    public List<Task> findMatchedTasks(String... keywords) {
+        Stream<Task> matchedTasksStream = Stream.empty();
+        for (String keyword : keywords) {
+            if (keyword.isEmpty()) {
+                continue;
+            }
+            matchedTasksStream = this.tasks.stream()
+                    .filter((task) -> task.getDescription().contains(keyword));
+        }
+        return matchedTasksStream.collect(Collectors.toList());
+    }
+}
```

Comment:
I like how you use the stream idea to search, leading to a very concise code and readable !